### PR TITLE
ci: Remove Jenkins Prechecks

### PR DESCRIPTION
**Pull request by nathanjsweet**
Now that we have a github workflow that does the
static language prechecks on both pull-requests and
the master branch we no longer need to run these checks
inside Jenkins.

```release-note
Removed unnecessarily redundant static analysis in CI to streamline CI running times.
```

### CODE REVIEW

**Comment by nathanjsweet** [comment]
test-me-please

**Review by christarazi — APPROVED** [review]
:tada: Never knew we had this in Jenkins.

File: test/docker-compose.yml [hunk 1]
```diff
@@ -19,7 +19,3 @@ services:
     volumes:
       - "./../:/go/src/github.com/cilium/cilium/"
     privileged: true
-  precheck:
-    extends:
-      service: base_image
-    command: "bash -c 'cd /go/src/github.com/cilium/cilium/; make ci-precheck || exit 1'"
```
Comment (tklauser) [inline test/docker-compose.yml:25]: Seems this is the only usage of the `make ci-precheck` target, so it could be removed as well while at it.

**Review by tklauser — COMMENTED** [review]
One remark regarding `make ci-precheck`, otherwise LGTM.

File: test/docker-compose.yml (hunk 1 above)
Comment (pchaigno) [reply test/docker-compose.yml:25]: We can also remove this `test/docker-compose.yml` file by the looks of it.

**Review by nebril — CHANGES_REQUESTED** [review]
Additional removals suggested by @pchaigno and @tklauser would be perfect.

**Review by pchaigno — APPROVED** [review]

**Comment by nathanjsweet** [comment]
test-me-please

**Review by nebril — APPROVED** [review]

**Review by tklauser — APPROVED** [review]

**Pull request merged**